cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/security/jwt/JwtUserIdentity.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2020 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.servlet.security.jwt;
+
+import org.eclipse.jetty.server.UserIdentity;
+
+import javax.security.auth.Subject;
+import java.security.Principal;
+
+public class JwtUserIdentity implements UserIdentity {
+
+  private Subject _subject;
+  private Principal _principal;
+  private UserIdentity _roleDelegate;
+
+  JwtUserIdentity(Subject subject, Principal principal, UserIdentity roleDelegate) {
+    _subject = subject;
+    _principal = principal;
```

Comment:
Is there an implicit assumption that `subject` and `roleDelegate` cannot be `null`? -- e.g. `JwtUserIdentity#getSubject()` is accessed w/o a `null` check, which may potentially yield an `NPE`?

Reply:
Good catch. In fact it could happen that in `JwtLoginService.login` `null` is returned by the 
```UserIdentity rolesDelegate = _authorizationService.getUserIdentity((HttpServletRequest) request, username);``` 
call. In this case we should return with an unauthenticated identity and handle that case as an unsuccessful login. Otherwise if the login was successful then the `subject` and `roleDelegate` shouldn't be `null`.